### PR TITLE
Fix typos in developer-guide.adoc

### PR DESCRIPTION
**Pull request by LackOfMorals**
Corrected a number of typos in the developer-guide.adoc pages

### CODE REVIEW

File: docs/modules/ROOT/pages/developer-guide.adoc [hunk 1]
```diff
@@ -10,7 +10,7 @@ to activate the virtual environment.
 
 Next, run:
 ----
-pip install --editable .
+pip install --editable
```
Comment (danielruminski) [inline docs/modules/ROOT/pages/developer-guide.adoc:13]: This one is not a spelling mistake, the `.` indicates that you want to install the current directory

**Review by danielruminski — CHANGES_REQUESTED** [review]
Thanks for the fixes! One of them needs to be reversed though.

File: docs/modules/ROOT/pages/developer-guide.adoc (hunk 1 above)
Comment (LackOfMorals) [reply docs/modules/ROOT/pages/developer-guide.adoc]: Done, reversed it

**Review by danielruminski — APPROVED** [review]
LGTM!

**Review by LackOfMorals — COMMENTED** [review]
👍

**Pull request merged**